ogre2/src/Ogre2RenderTarget.cc

```diff
@@ -346,6 +346,17 @@ void Ogre2RenderTarget::UpdateShadowNode()
     }
   }
 
+  // limit number of shadow maps
+  // shaders dynamically generated by ogre produce compile error at runtime if
+  // the number of shadow maps exceeds certain number. The error seems to
+  // suggest that the number of uniform variables has exceeded the max number
+  // allowed
+  unsigned int maxShadowMaps = 25u;
+  dirLightCount = std::min(static_cast<unsigned int>(maxShadowMaps / 3),
+      dirLightCount);
+  spotPointLightCount = std::min(std::max(maxShadowMaps - dirLightCount*3u, 0u),
+      spotPointLightCount);
```

Comment:
This looks like something that may confuse users down the line. How about printing a warning when the maximum number is reached? I think the warning should only be printed once.

Reply:
yep added warning msg in 7b84199